numpy/lib/arraypad.py

```diff
@@ -986,399 +136,191 @@ def _validate_lengths(narray, number_elements):
     return normshp
 
 
-###############################################################################
-# Public functions
-
-
-def pad(array, pad_width, mode, **kwargs):
+def _arange_ndarray(shape, axis, reverse=False):
```

Comment:
In the interest of minimizing the diff, can you leave this function where it originally was?